aws/modules/vpc/main.tf

```diff
@@ -0,0 +1,219 @@
+    }
+  )
+}
+
+#----------------------------------------------------------------------
+resource "aws_eip" "mod_nat_eip" {
+  count = var.multi_az_nat_gateway * local.pri_az_count + var.single_nat_gateway * 1
+  tags  = var.tags
+  vpc   = true
+}
+
+#-------------------------------------------------------------------------
+
+
+data "http" "workstation-external-ip" {
+  url = "http://ipv4.icanhazip.com"
+}
+
+data "aws_availability_zones" "availability_zones" {
+}
+
+resource "aws_security_group" "security_group" {
+  name        = format("%s-security-group", var.name)
+  description = "Cluster communication with worker nodes"
+  vpc_id      = aws_vpc.vpc.id
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = var.tags
+}
+
+resource "aws_security_group_rule" "https_security_group_rule" {
+  description              = "Allow pods to communicate with the cluster API Server"
+  from_port                = 443
+  protocol                 = "tcp"
+  security_group_id        = aws_security_group.security_group.id
+  source_security_group_id = aws_security_group.worker_security_group.id
+  to_port                  = 443
+  type                     = "ingress"
+}
+
+resource "aws_security_group_rule" "workstation_https_group_rule" {
+  cidr_blocks       = [local.workstation-external-cidr]
+  description       = "Allow workstation to communicate with the cluster API Server"
+  from_port         = 443
+  protocol          = "tcp"
+  security_group_id = aws_security_group.security_group.id
+  to_port           = 443
+  type              = "ingress"
+}
+
+# workers
+resource "aws_security_group" "worker_security_group" {
+  name        = format("%s-worker-security-group", var.name)
+  description = "Security group for all nodes in the cluster"
+  vpc_id      = aws_vpc.vpc.id
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = var.tags
+}
+
+resource "aws_security_group_rule" "self_security_group_rule" {
+  description              = "Allow node to communicate with each other"
+  from_port                = 0
+  protocol                 = "-1"
+  security_group_id        = aws_security_group.worker_security_group.id
+  source_security_group_id = aws_security_group.worker_security_group.id
+  to_port                  = 65535
+  type                     = "ingress"
+}
+
+resource "aws_security_group_rule" "cluster_security_group_rule" {
+  description              = "Allow worker Kubelets and pods to receive communication from the cluster control plane"
+  from_port                = 1025
+  protocol                 = "tcp"
+  security_group_id        = aws_security_group.worker_security_group.id
+  source_security_group_id = aws_security_group.worker_security_group.id
+  to_port                  = 65535
+  type                     = "ingress"
+}
+
+provider "aws" {
+  region = "us-east-1"
```

Comment:
Variable please

Reply:
I have used this data block for internal testing. this data block won't be here. it will be on the env/versions.tf folder. will change it to the variable.